ytmusicapi/parsers/playlists.py

```diff
@@ -44,15 +44,13 @@ def parse_playlist_items(results, menu_entries: Optional[List[List]] = None, is_
         if title == "Song deleted":
             continue
 
+        flex_column_count = len(data["flexColumns"])
+
         artists = parse_song_artists(data, 1)
 
-        album = parse_song_album(data, 2)
+        album = parse_song_album(data, flex_column_count-1) if not is_album else None
 
-        views = None
-        if album and album["id"] is None:
-            # views currently only present on albums and formatting is localization-dependent -> no parsing
-            if (views := (get_item_text(data, 2))) is not None:
-                album = None
+        views = get_item_text(data, 2) if flex_column_count == 4 or is_album else None
```

Comment:
> flex_column_count == 4 or is_album

which cases does this condition cover? `is_album` is obvious, but when does the `count==4` apply when it's not an album?

Reply:
I am covering
- Album with three columns, the last of which is views/plays:
![grafik](https://github.com/sigma67/ytmusicapi/assets/82546407/668a0e91-7c05-437e-8b89-cae2dcf0f1f3)
- Playlist with three columns, the last of which is album:
![grafik](https://github.com/sigma67/ytmusicapi/assets/82546407/da9d345a-c8ca-4688-8478-6866fdd5d760)
- Artist songs from Artist overview with four columns, the third of which is views/plays: and the fourth of which is album:
![grafik](https://github.com/sigma67/ytmusicapi/assets/82546407/3c4e51be-d861-4367-b354-e2eed132255f)
- Artist song details with four columns, the third of which is views/plays: and the fourth of which is album:
![grafik](https://github.com/sigma67/ytmusicapi/assets/82546407/87da453b-4d53-4466-8a67-4f6ba81a42c0)

Reply:
Thanks for this overview, that explains it!